cmd/juno/juno.go

```diff
@@ -202,8 +224,25 @@ func NewCmd(config *node.Config, run func(*cobra.Command, []string) error) *cobr
 
 		// TextUnmarshallerHookFunc allows us to unmarshal values that satisfy the
 		// encoding.TextUnmarshaller interface (see the LogLevel type for an example).
-		return v.Unmarshal(config, viper.DecodeHook(mapstructure.ComposeDecodeHookFunc(
-			mapstructure.TextUnmarshallerHookFunc(), mapstructure.StringToTimeDurationHookFunc())))
+		if err := v.Unmarshal(config, viper.DecodeHook(mapstructure.ComposeDecodeHookFunc(
+			mapstructure.TextUnmarshallerHookFunc(), mapstructure.StringToTimeDurationHookFunc()))); err != nil {
+			return err
+		}
+
+		// Set custom network
+		if config.Network.String() == "custom" {
+			if v.GetString(cnFeederURLF) == "" {
+				return errors.New("custom networks must specify all `cn` flags (--cn-feeder-url etc)")
```

Comment:
Can we remove these checks and make `--network` mutually exclusive with `--cn-feeder-url` instead? Then we can allow for custom network names, too.

Reply:
In the current implementation, we require network to be set to "custom" if users want to run a custom network, so these can't be made mutually exclusive. Wasn't sure how to do this in a neater way beyond creating a new function etc

Reply:
Update: I didn't originally understand, but I'm updated the code so that these flags are mutually exclusive.